transform.js

```diff
@@ -39,23 +50,40 @@ function transform (filename, opts) {
 
     // transform all detected nodes and
     // close stream when done
-    mapLimit(nodes, Infinity, iterate, function () {
+    mapLimit(nodes, Infinity, iterate, function (err) {
+      if (err) return this.emit('error', err)
       self.push(ast.toString())
       self.push(null)
     })
 
     // asynchronously update css and apply to node
+    // or exorcise to file
     function iterate (args, done) {
       const transform = args[0]
       const node = args[1]
       transform(function (err, css, prefix) {
         if (err) return done(err)
-        const str = [
-          "((require('insert-css')(" + JSON.stringify(css) + ')',
-          ' || true) && ' + JSON.stringify(prefix) + ')'
-        ].join('')
-        node.update(str)
-        done()
+        if (opts.out) {
+          // exorcise to external file
+          const dirname = path.dirname(opts.out)
+          mkdirp(dirname, function (err) {
+            if (err) return done(err)
+            const ws = (typeof opts.out === 'string')
+              ? fs.createWriteStream(opts.out)
+              : opts.out
+            eos(ws, done)
+            node.update('0')
+            ws.end(css)
```

Comment:
yeah, that would be correct - but given that to my understanding the transform happens in sequence (e.g. next chunk isn't parsed until this one is done) that should not be an issue... I hope?